test/any/parsing/BaseParser-test.js

```diff
@@ -0,0 +1,55 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
+describe('BaseParser', function() {
+  beforeEach(function() {
+    foam.CLASS({
+      name: 'TestParser',
+      extends: 'org.chromium.webidl.BaseParser',
+
+      constants: {
+        PARSE_COMPLETE: 'Parse complete',
+        PARSE_INCOMPLETE: 'Parse incomplete'
+      },
+
+      properties: [
+        {
+          name: 'separator',
+          factory: function() {
+            return this.Parsers.create().literal(' ');
+          }
+        }
+      ],
+
+      methods: [
+        function symbolsFactory() {
+          return Object.assign(this.SUPER(), foam.Function.withArgs(
```

Comment:
Since we don't use anything inherited from `symbolsFactory` let's just make this explicit:

(1) Add `implements: ['foam.parse.Parsers']` to class; this will mixin its methods;
(2) Use:

```js
return {START: this.literal('interface Frobinator;')};
```
Similarly for `separator` above:

```js
return this.literal(' ');
```